mediorum/server/signature/signature.go

```diff
@@ -120,3 +121,55 @@ func recover(input string, signature []byte) (common.Address, error) {
 	hash := crypto.Keccak256Hash([]byte(input))
 	return sigverify.EcRecoverEx(hash.Bytes(), signature)
 }
+
+func sortedMap(m map[string]interface{}) map[string]interface{} {
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	result := make(map[string]interface{})
+	for _, k := range keys {
+		switch v := m[k].(type) {
+		case map[string]interface{}:
+			result[k] = sortedMap(v)
+		case []interface{}:
+			arr := make([]interface{}, len(v))
+			for i, elem := range v {
+				if elemMap, ok := elem.(map[string]interface{}); ok {
+					arr[i] = sortedMap(elemMap)
+				} else {
+					arr[i] = elem
+				}
+			}
+			result[k] = arr
+		default:
+			result[k] = v
+		}
+	}
+
+	return result
+}
+
+func SortKeys(data interface{}) (string, error) {
+	jsonString, err := json.Marshal(data)
```

Comment:
I think marshal may sort?
https://pkg.go.dev/encoding/json#Marshal

```
Map values encode as JSON objects. The map's key type must either be a string, an integer type, or implement encoding.TextMarshaler. The map keys are sorted and used as JSON object keys by applying the following rules, subject to the UTF-8 coercion described for string values above:
```

Not a blocker, but hand-rolling this code seems unnecessary / potentially error prone

Reply:
Or https://github.com/golang/go/commit/a440cc0d702e15c19bcc984f7a8f5c10f83726ab#diff-c9a1d96819ff52bfce0eb0e282fc3321

Reply:
Some interesting reading 
https://www.reddit.com/r/golang/comments/9y7nw2/til_jsonmarshal_sorts_map_keys/

Reply:
ah I see, so it sounds like `fmt` sorts, but I think it only does maps and not structs. I debugged the test I made, and it's showing unsorted after `json.Marshal` when I pass in the health check struct. afaik no consumer actually verifies the health check signature, but if someone ever did then I think it has to keep this sorting to be deterministic

Reply:
looks like there's a simpler way after all! after marshalling, the test passes when just doing `jcs.Transform` (from https://github.com/gowebpki/jcs)

Reply:
👑